### PR TITLE
[FIX] product_pricelist_supplierinfo: Fix broken get price with False date parameter

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT28603

ping @carlosdauden @CarlosRoca13

### CODE REVIEW

File: product_pricelist_supplierinfo/models/product_template.py [hunk 1]
```diff
@@ -25,10 +25,11 @@ def _get_supplierinfo_pricelist_price(
         if product:
             if type(date) == datetime:
                 date = date.date()
+            # Fails is date is False so pass None
```
Comment (pedrobaeza) [inline product_pricelist_supplierinfo/models/product_template.py:28]:
```suggestion
            # Fails if date is False, so pass None
```

**Comment by pedrobaeza** [comment]
Isn't this an error in the module that passes False instead of None?

**Comment by carlosdauden** [comment]
Maybe we should change this:
https://github.com/OCA/product-attribute/blob/95d0805600cbfa21b32550f2fa760ffd4e5b53b3/product_pricelist_supplierinfo/models/product_pricelist.py#L29
to this other:
`date=date or context.get('date') or fields.Date.today(),`

**Comment by pedrobaeza** [comment]
Yeah, if that is the problem, there's a chance of having `date` key in the context, but containing `False`/`None`.

**Review by chienandalu — COMMENTED** [review]
I'd vote for @carlosdauden suggestion

**Pull request closed without merging**